src/api.js

```diff
@@ -0,0 +1,797 @@
+    rowObject = {};
+    for (i = k = 0, len = names.length; k < len; i = ++k) {
+        name = names[i];
+        rowObject[name] = values[i];
+    }
+    return rowObject;
+};
+
+/* Shorthand for bind + step + reset
+Bind the values, execute the statement, ignoring the rows it returns, and resets it
+@param [Array,Object] Value to bind to the statement
+ */
+Statement.prototype["run"] = function(values) {
+    if (values != null) {
+        this["bind"](values);
+    }
+    this["step"]();
+    return this["reset"]();
+};
+
+Statement.prototype.bindString = function(string, pos) {
+    var bytes, strptr;
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    bytes = intArrayFromString(string);
+    this.allocatedmem.push(strptr = allocate(bytes, "i8", ALLOC_NORMAL));
+    this.db.handleError(sqlite3_bind_text(this.stmt, pos, strptr, bytes.length - 1, 0));
+    return true;
+};
+
+Statement.prototype.bindBlob = function(array, pos) {
+    var blobptr;
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    this.allocatedmem.push(blobptr = allocate(array, "i8", ALLOC_NORMAL));
+    this.db.handleError(sqlite3_bind_blob(this.stmt, pos, blobptr, array.length, 0));
+    return true;
+};
+
+Statement.prototype.bindNumber = function(num, pos) {
+    var bindfunc;
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    bindfunc = num === (num | 0) ? sqlite3_bind_int : sqlite3_bind_double;
+    this.db.handleError(bindfunc(this.stmt, pos, num));
+    return true;
+};
+
+Statement.prototype.bindNull = function(pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    return sqlite3_bind_blob(this.stmt, pos, 0, 0, 0) === SQLITE_OK;
+};
+
+Statement.prototype.bindValue = function(val, pos) {
+    if (pos == null) {
+        pos = this.pos++;
+    }
+    switch (typeof val) {
+    case "string":
+        return this.bindString(val, pos);
+    case "number":
+    case "boolean":
+        return this.bindNumber(val + 0, pos);
+    case "object":
+        if (val === null) {
+            return this.bindNull(pos);
+        } else if (val.length != null) {
+            return this.bindBlob(val, pos);
+        } else {
+            throw "Wrong API use : tried to bind a value of an unknown type (" + val + ").";
+        }
+    }
+};
+
+/* Bind names and values of an object to the named parameters of the statement
+@param [Object]
+@private
+@nodoc
+ */
+Statement.prototype.bindFromObject = function(valuesObj) {
+    var name, num, value;
+    for (name in valuesObj) {
+        value = valuesObj[name];
+        num = sqlite3_bind_parameter_index(this.stmt, name);
+        if (num !== 0) {
+            this.bindValue(value, num);
+        }
+    }
+    return true;
+};
+
+/* Bind values to numbered parameters
+@param [Array]
+@private
+@nodoc
+ */
+Statement.prototype.bindFromArray = function(values) {
+    var k, len, num, value;
+    for (num = k = 0, len = values.length; k < len; num = ++k) {
```

Comment:
we need only 1 variable

Reply:
implemented suggestion in commit 0d78adb79ee694e45bb4223b5fc2ebc99b45ebec